chain/db/table.go

```diff
@@ -47,6 +47,24 @@ type DataSourceRevision struct {
 	TxHash         []byte `gorm:"not null"`
 }
 
+type OracleScript struct {
+	ID          int64  `gorm:"primary_key;auto_increment:false"`
+	Name        string `gorm:"not null"`
+	Description string `gorm:"not null"`
+	Owner       string `gorm:"not null"`
+	Code        []byte `gorm:"not null"`
+	LastUpdated int64  `gorm:"not null"`
+}
+
+type OracleScriptRevision struct {
+	OracleScriptID int64  `gorm:"primary_key;auto_increment:false"`
+	RevisionNumber int64  `gorm:"primary_key"`
+	Name           string `gorm:"not null"`
+	Timestamp      int64  `gorm:"not null"`
+	BlockHeight    int64  `gorm:"not null"`
+	TxHash         []byte `gorm:"not null"`
```

Comment:
Add fk to transaction

Reply:
@taobun should we add fk here? It will not work with oracle scripts generated during genesis......

Reply:
If we allow to have txHash 000000... as genesis state, it sounds make sense?

Reply:
Yeah let's do that.

Reply:
Three choices:
- Don't do FK
- Use dummy FK like you suggest
- Make `txHash` column nullable. NULL does not need to match FK.

Reply:
The answer is Make txHash column nullable. NULL does not need to match FK.

Reply:
watch #995 as example